tests/local/mod_accredible_formhelper_test.php

```diff
@@ -0,0 +1,140 @@
+        }
+        $this->coursegradeitemid = $DB->get_field('grade_items', 'id', $gradeitem);
+    }
+
+    /**
+     * Test the load_grade_item_options method.
+     * @covers ::load_grade_item_options
+     */
+    public function test_load_grade_item_options() {
+        $formhelper = new formhelper();
+
+        // When there are no grade items.
+        $expected = array(
+            '' => 'Select an Activity Grade',
+            $this->coursegradeitemid => get_string('coursetotal', 'accredible')
+        );
+        $result = $formhelper->load_grade_item_options($this->course->id);
+        $this->assertEquals($expected, $result);
+
+        // When there are grade items.
+        $quizid1 = $this->create_quiz_module($this->course->id, 'Quiz 1');
+        $gradeitemid1 = $this->create_grade_item($this->course->id, 'Quiz 1', 'quiz', $quizid1);
+
+        $quizid2 = $this->create_quiz_module($this->course->id, 'Quiz 2');
+        $gradeitemid2 = $this->create_grade_item($this->course->id, 'Quiz 2', 'quiz', $quizid2);
+
+        $expected = array(
+            '' => 'Select an Activity Grade',
+            $this->coursegradeitemid => get_string('coursetotal', 'accredible'),
+            $gradeitemid1 => 'Quiz 1',
+            $gradeitemid2 => 'Quiz 2',
+        );
+        $result = $formhelper->load_grade_item_options($this->course->id);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * fetch course grate item record
+     *
+     * @param int $courseid
+     */
+    private function fetch_course_grade_item($courseid) {
+        global $DB;
+
+        return $DB->get_record(
+            'grade_items',
+            array(
+                'courseid' => $courseid,
+                'itemtype' => 'course',
+                'itemmodule' => null
+            ),
+            '*',
+            MUST_EXIST
+        );
+    }
+
+    /**
+     * Create quiz module test
+     *
+     * @param int $courseid
+     * @param string $name
+     */
+    private function create_quiz_module($courseid, $name) {
+        global $DB;
+
+        return $DB->insert_record('quiz',
+            array(
+                'course' => $courseid,
+                'name' => $name,
+                'intro' => 'Default intro',
+                'grade' => 10
+            )
+        );
+    }
+
+    /**
+     * Create quiz module test
+     *
+     * @param int $courseid
+     * @param string $itemname
+     * @param string $itemmodule
+     * @param int $iteminstance
+     */
+    private function create_grade_item($courseid, $itemname, $itemmodule, $iteminstance) {
+        global $DB;
+        $gradeitem = array(
+            "courseid" => $courseid,
+            "itemname" => $itemname,
+            "itemtype" => 'mod',
+            "itemmodule" => $itemmodule,
+            "iteminstance" => $iteminstance,
+            "itemnumber" => 0
+        );
+        return $DB->insert_record('grade_items', $gradeitem);
```

Comment:
The same, `getDataGenerator` doesn't work as intended so it's necessary to use `insert_record` here.